compiler/control/OMROptions.cpp

```diff
@@ -2407,7 +2410,7 @@ OMR::Options::processOptionsJIT(char *jitOptions, void *feBase, TR_FrontEnd *fe)
       }
 
    if (_jitCmdLineOptions)
-      memset(_jitCmdLineOptions, 0, sizeof(TR::Options));
+      memset(_jitCmdLineOptions->_options, 0, sizeof(_options));
```

Comment:
Before this change the code was initializing the entire Options object to 0 (thus, all fields were set to 0).
After this change, only the `_options` array is initialized to 0 and the other fields may receive random values.
We need to keep this line or write a full constructor that initializes all fields to 0, not just the `_options` array.

Reply:
Yup, I can confirm that it's not enough to just initialize the _options array or even to use the default constructor. If we don't want to use memset we'll have to create a new constructor as you say.

The default constructor is actually used in the other branch of this if statement:
https://github.com/eclipse/omr/blob/a57ad601d6664d97958c59884c4d34a0ce2f960a/compiler/control/OMROptions.cpp#L2405
and it seems to work perfectly fine, but given how finnicky this bug was, I'm not sure if it's intentional or just a fluke.

Reply:
This memset is also not needed, because the contructor does it now inside.

Reply:
Do we not need to reset the _options array to all 0's? I thought that was the original purpose of using memset there in the first place. Otherwise there would have been no need for the `if(_jitCmdLineOptions)` branch at all

Reply:
The constructor with no arguments TR::Options() initializes the options array:
`memset(_options, 0, sizeof(_options));`.
Because of this, when we execute: `_jitCmdLineOptions = new (PERSISTENT_NEW) TR::Options();` all fields are correctly initialized to 0, including the _options array.

Reply:
I am also not sure about this code. If `processOptionsJIT `is called only once , we don't need that `memset`. If it's called more than once and we would like to reinitialize the whole `OMROptions` object why do we only reinitialize `_options[]` ?

Reply:
So address the possibility of `processOptionsJIT()` being called more than once and perserve the original intended functionality of this part of the code, I have created an `init()` function for `TR::Options` that can be used both by the default constructor and in `procesOptionsJIT()` when `_jitCmdLineOptions` has already been allocated but needs to be reinitialized.